mlir/tools/mlir-tblgen/DialectGen.cpp

```diff
@@ -341,7 +349,17 @@ static const char *const dialectDestructorStr = R"(
 {0}::~{0}() = default;
 
 )";
+  
+/// The code block to generate a member funcs.
+///
+/// {0}: The name of the dialect class.
+static const char *const dialectStaticMemberDefs = R"(
+const {0} *{0}::getLoaded(::mlir::Operation *operation) {{
```

Comment:
I'm not convinced of the utility of this method over just casting from the operations dialect. This method is going to be significantly slower than calling getDialect on the operation itself. If we wanted to make this easier, we should rather consider auto-generating a templated version of `getDialect` for Ops/Attrs/Types that does the casting for you.

Reply:
This also seems like an unrelated change compared to the description of this commit.

Reply:
The dialect of the operation may not be the same dialect that owns the attribute though.

Reply:
This method seems ad-hoc to me and does not provide much convenience compared to the current code.
What's wrong with `getContext().getLoadedDialect<ROCDL::ROCDLDialect>();` ?

Reply:
Nothing really, just adding for convenience. I was asked on a dialect in a separate project to make usage of AttrHelpers cleaner.

Reply:
This particular method was added in the cpp file gen so Operation.h would not require inclusion by all Dialects. But it is just an extra convenience, happy to remove.